controllers/cloud.redhat.com/clowdjobinvocation_controller.go

```diff
@@ -173,102 +212,244 @@ func (r *ClowdJobInvocationReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 
 // InvokeJob is responsible for applying the Job. It also updates and reports
 // the status of that job
-func (r *ClowdJobInvocationReconciler) InvokeJob(ctx context.Context, job *crd.Job, app *crd.ClowdApp, env *crd.ClowdEnvironment, cji *crd.ClowdJobInvocation) error {
+func (r *ClowdJobInvocationReconciler) InvokeJob(cache *providers.ObjectCache, job *crd.Job, app *crd.ClowdApp, env *crd.ClowdEnvironment, cji *crd.ClowdJobInvocation) error {
 	nn := types.NamespacedName{
 		Name:      fmt.Sprintf("%v-%v-%v", app.Name, job.Name, cji.Name),
 		Namespace: app.Namespace,
 	}
-	j := batchv1.Job{}
 
-	createJobResource(cji, env, nn, job, &j)
-	if err := r.Client.Create(ctx, &j); err != nil {
+	var ClowdJob = providers.NewMultiResourceIdent("clowdjob", "core_clowdjob", &crd.ClowdJobInvocation{})
+
+	j := batchv1.Job{}
+	jobProvider.CreateJobResource(cji, env, nn, job, &j)
+	if err := cache.Create(ClowdJob, nn, &j); err != nil {
 		return err
 	}
+
 	cji.Status.Jobs = append(cji.Status.Jobs, j.ObjectMeta.Name)
 	r.Log.Info("Job Invoked Successfully", "jobinvocation", job.Name, "namespace", app.Namespace)
 	r.Recorder.Eventf(cji, "Normal", "ClowdJobInvoked", "Job [%s] was invoked successfully", j.ObjectMeta.Name)
 
 	return nil
 }
 
-// applyJob build the k8s job resource and applies it from the Job config
-// defined in the ClowdApp
-func createJobResource(cji *crd.ClowdJobInvocation, env *crd.ClowdEnvironment, nn types.NamespacedName, job *crd.Job, j *batchv1.Job) {
+func (r *ClowdJobInvocationReconciler) fetchConfig(name types.NamespacedName, ctx context.Context) (config.AppConfig, error) {
+
+	secretConfig := core.Secret{}
+	jsonContent := config.AppConfig{}
+
+	err := r.Client.Get(ctx, name, &secretConfig)
+
+	if err != nil {
+		return jsonContent, err
```

Comment:
Personally I would prefer to ensure we send back `config.AppConfig{}`, rather than `jsonContent` - that way we can ensure it's null in the case of an error. Also, jsonContent isn't storing jsonContent, it's a struct, could be a little misleading.